code/modules/nifsoft/nif.dm

```diff
@@ -1,625 +1,625 @@
+
+/obj/item/device/nif/proc/del_plane(var/planeid = null)
+	if(!planeid)
+		return
+	planes_visible -= planeid
+
+/obj/item/device/nif/proc/vis_update()
+	if(human)
+		human.recalculate_vis()
+
+// Alternate NIFs
+/obj/item/device/nif/bad
+	name = "bootleg NIF"
+	desc = "A copy of a copy of a copy of a copy of... this can't be any good, right? Surely?"
+	durability = 10
+
+/obj/item/device/nif/authentic
+	name = "\improper Kitsuhana NIF"
+	desc = "An actual Kitsuhana working surface, in a box. From a society slightly less afraid \
+	of self-replicating nanotechnology. Basically just a high-endurance NIF."
+	durability = 1000
+
+/obj/item/device/nif/bioadap
+	name = "bioadaptive NIF"
+	desc = "A NIF that goes out of it's way to accomidate strange body types. \
+	Will function in species where it normally wouldn't."
+	durability = 25
+	bioadap = TRUE
+
+////////////////////////////////
+// Special Promethean """surgery"""
+/obj/item/device/nif/attack(mob/living/M, mob/living/user, var/target_zone)
+	if(!ishuman(M) || !ishuman(user) || (M == user))
+		return ..()
+
+	var/mob/living/carbon/human/U = user
+	var/mob/living/carbon/human/T = M
+
+	if(istype(T.species,/datum/species/shapeshifter/promethean) && target_zone == BP_TORSO)
+		if(T.w_uniform || T.wear_suit)
+			to_chat(user,"<span class='warning'>Remove any clothing they have on, as it might interfere!</span>")
+			return
+		var/obj/item/organ/external/eo = T.get_organ(BP_TORSO)
+		if(!T)
+			to_chat(user,"<span class='warning'>They should probably regrow their torso first.</span>")
+			return
+		U.visible_message("<span class='notice'>[U] begins installing [src] into [T]'s chest by just stuffing it in.</span>",
+		"<span class='notice'>You begin installing [src] into [T]'s chest by just stuffing it in.</span>",
+		"There's a wet SQUISH noise.")
+		if(do_mob(user = user, target = T, time = 200, target_zone = BP_TORSO))
+			user.unEquip(src)
+			forceMove(eo)
+			eo.implants |= src
+			implant(T)
+			playsound(T,'sound/effects/slime_squish.ogg',50,1)
+	else
+		return ..()
+
+/mob/living/carbon/human/proc/set_nif_examine()
+	set name = "NIF Appearance"
+	set desc = "If your NIF alters your appearance in some way, describe it here."
+	set category = "OOC"
+
+	if(!nif)
+		verbs -= /mob/living/carbon/human/proc/set_nif_examine
+		to_chat(src,"<span class='warning'>You don't have a NIF, not sure why this was here.</span>")
+		return
+
+	var/new_flavor = sanitize(input(src,"Describe how your NIF alters your appearance, like glowy eyes or metal plate on your head, etc. Be sensible. Clear this for no examine text. 128ch max.","Describe NIF", nif.examine_msg) as null|text, max_length = 128)
+	//They clicked cancel or meanwhile lost their NIF
+	if(!nif || isnull(new_flavor))
+		return //No changes
+	//Sanitize or user cleaned it entirely
+	if(!new_flavor)
+		nif.examine_msg = ""
+		nif.save_data["examine_msg"] = ""
+	else
+		nif.examine_msg = new_flavor
+		nif.save_data["examine_msg"] = new_flavor

```

Comment:
??? this entire file what happened

Reply:
Apparently some whitespace disappeared from two blank lines among other things. I'll fix that.